apps/zui/scripts/merge-mac-release-files.mjs

```diff
@@ -0,0 +1,226 @@
+  }
+
+  const localLatestMacYmlContent = fs.readFileSync(LOCAL_FILE_PATH, { encoding: 'utf8' });
+
+  const localPlatform = getPlatformFromLatestMacYml(localLatestMacYmlContent);
+
+  if (localPlatform === 'none' || localPlatform === 'both') {
+    console.log(`[local] ${FILE_NAME} invalid. Platform: ${localPlatform}. Skipping merge`)
+    return;
+  } else {
+    console.log(`[local] ${FILE_NAME} valid: Platform: ${localPlatform}`)
+  }
+
+  const localPlatformPresentRemotely = currentRelease.assets.find(asset => {
+    return asset.name === `latest-mac-${localPlatform}.yml`;
+  });
+
+  if (localPlatformPresentRemotely) {
+    try {
+      await client.request(`DELETE ${URL}/assets/${localPlatformPresentRemotely.id}`);
+      console.log(`[remote] deleted latest-mac-${localPlatform}.yml`)
+    } catch(e) {
+      console.log(`[remote] error deleting latest-mac-${localPlatform}.yml. Skipping merge`)
+      console.log(e);
+      return;
+    }
+  }
+
+  const uploadUrl = currentRelease.upload_url;
+  const localAssetStream = new Readable();
+  localAssetStream.push(localLatestMacYmlContent);
+  localAssetStream.push(null);
+
+  try {
+    await client.rest.repos.uploadReleaseAsset({
+      url: uploadUrl,
+      headers: {
+        'content-type': 'application/octet-stream',
+        'content-length': Buffer.byteLength(localLatestMacYmlContent),
+      },
+      name: `latest-mac-${localPlatform}.yml`,
+      data: localAssetStream,
+    })
+    console.log(`[remote] latest-mac-${localPlatform}.yml uploaded`)
+  } catch(e) {
+    console.log(`[remote] error uploading latest-mac-${localPlatform}.yml. Skipping merge`);
+    console.log(e);
+    return;
+  }
+
+  const remotePlatform = localPlatform === 'intel' ? 'arm' : 'intel';
+
+  const remotePlatformFileExists = currentRelease.assets.find(asset => {
+    return asset.name === `latest-mac-${remotePlatform}.yml`;
+  })
+
+  if (!remotePlatformFileExists) {
+    console.log(`[remote] latest-mac-${remotePlatform}.yml does not exist. Skipping merge`)
+    return;
+  } else {
+    console.log(`[remote] latest-mac-${remotePlatform}.yml found`)
+  }
+
+  let remotePlatformFile = null;
+
+  try {
+    remotePlatformFile = await client.request(`GET ${URL}/assets/${remotePlatformFileExists.id}`, {
+      headers: {
+        accept: 'application/octet-stream'
+      }
+    });
+    console.log(`[remote] latest-mac-${remotePlatform}.yml downloaded`)
+  } catch(e) {
+    console.log(`[remote] error downloading latest-mac-${remotePlatform}.yml. Skipping merge`);
+    console.log(e);
+    return;
+  }
+
+  const remoteLatestMacYmlContent = new TextDecoder().decode(remotePlatformFile.data);
+
+  try {
+    const originalAsset = currentRelease.assets.find(asset => {
+      return asset.name === FILE_NAME;
+    })
+
+    if (!originalAsset) {
+      console.log(`[remote] ${FILE_NAME} not found. Skipping merge`);
+      return;
+    } else {
+      console.log(`[remote] ${FILE_NAME} found`)
+    }
```

Comment:
I would move this stuff out of the try block since that it was you are doing elsewhere in this file. Also you can drop the else and just console log.